CantEngine/Systems/SplineCurvesSystem.cpp

```diff
@@ -0,0 +1,74 @@
+#include "stdafx.h"
+#include "SplineCurvesSystem.h"
+#include "GameObjects/GameObject.h"
+#include "Components/TransformComponent.h"
+#include "Components/SplineCurvesComponent.h"
+#include "Graphics/AppRenderer.h"
+#include "Graphics/DebugRendering/DebugRendering.h"
+
+
+unsigned const SplineCurvesSystem::static_type = BaseSystem::numberOfTypes++;
+
+SplineCurvesSystem::SplineCurvesSystem()
+{
+	Push_required_comp<TransformComponent>();
+	Push_required_comp<SplineCurvesComponent>();
+}
+
+
+SplineCurvesSystem::~SplineCurvesSystem()
+{
+}
+
+
+void SplineCurvesSystem::Register_GameObject(GameObject *go)
+{
+	TransformComponent *transform = go->GetComponent<TransformComponent>();
+	SplineCurvesComponent* splineCurvesComp = go->GetComponent<SplineCurvesComponent>();
+
+	BaseSystemCompNode *component_node = new SplineCurvesCompNode(transform, splineCurvesComp);
+	component_node->m_goID = go->GetId();
+	this->m_ObjComponentsMap[go->GetId()] = component_node;
+}
+
+void SplineCurvesSystem::Update(float dt, BaseSystemCompNode *compNode)
+{
+
+}
+
+void SplineCurvesSystem::Draw(float dt, BaseSystemCompNode *compNode)
+{
+	auto splineCompNode = static_cast<SplineCurvesCompNode*>(compNode);
+	auto transform_comp = splineCompNode->m_transform;
+	auto curvesComp = splineCompNode->m_splineCurvesComp;
+
+	float steps = 0.01f;
+	Vector3 prevPosition = curvesComp->Evaluate(0.f);
+	for (float index = steps; index <= (1.f); index += steps)
```

Comment:
Are the register debug line and aabb for debugging? If so, just add developer guards. #ifdef